Libraries/LibWeb/CSS/Parser/SelectorParsing.cpp

```diff
@@ -70,8 +93,12 @@ Parser::ParseErrorOr<SelectorList> Parser::parse_a_selector_list(TokenStream<T>&
         auto stream = TokenStream(selector_parts);
         auto selector = parse_complex_selector(stream, mode);
         if (selector.is_error()) {
-            if (parsing_mode == SelectorParsingMode::Forgiving)
+            if (parsing_mode == SelectorParsingMode::Forgiving) {
+                // Keep the invalid selector around for serialization and nesting
+                auto combinator = mode == SelectorType::Standalone ? Selector::Combinator::None : Selector::Combinator::Descendant;
```

Comment:
This is the difference: We pass in a combinator to use, depending on what kind of selector we're parsing.

Passing the SelectorType might be better, but that involves more rearrangement because SelectorType is private to the Parser. 🤷